custom_components/sermatec_inverter/sensor.py

```diff
@@ -598,6 +597,42 @@ def _handle_coordinator_update(self) -> None:
         self._attr_native_value = abs(data) if data < 0 else 0
         self.async_write_ha_state()
 
+class SermatecBatteryChargingPowerSensor(SermatecSensor):
+    """
+    Special Sermatec sensor for battery charging power from current and voltage,
+    if actual battery_state is different "charging" return 0
+    value should always be negative for charging
+    ref: https://community.home-assistant.io/t/howto-fronius-integration-with-battery-into-energy-dashboard/376329
+    """
+
+    def __init__(self, coordinator, serial_number, dict_key, name, id = None, device_class = None, unit = None):
+        super().__init__(coordinator, serial_number, dict_key, name, id, device_class, unit)
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle data from the coordinator."""
+        data: int = int(self.coordinator.data[self.dict_key["voltage"]]) * int(self.coordinator.data[self.dict_key["current"]]) if self.coordinator.data["battery_state"] == "charging" else 0
+        self._attr_native_value = 0 - abs(data)
```

Comment:
The value can't be negative. The Home Assistant's energy dashboard expects a positive value for both total charged and discharged energy, which we provide by integrating (Riemann's sum) respective power sensors.

It also does not make much sense: If we have one power sensor, it can represent charging by a negative value and discharging by a positive value, here we have two sensors.

Reply:
done, I probably misread the doc ...